src/CoreLib/GV.fs

```diff
@@ -360,16 +361,520 @@ and /// Information of Within Job cluster information.
 
 
 
+/// Job Lifecycle management
+[<AllowNullLiteral>]
+type internal JobLifeCycle(jobID:Guid) = 
+    let CTSSource = new CancellationTokenSource()
+    let nExecutorOfDisposeJob = ref 0 
+    let nCancellationCalled = ref 0 
+    let onDispose = ExecuteUponOnce()
+    let onCancellation = ExecuteUponOnce()
+    let onException = ConcurrentQueue<Action<Exception>>()
+    let numExceptionCallback = ref 0
+    let exCollections = ConcurrentQueue<Exception>()
+    static member val MaxWaitToEndJobInMilliseconds = 30000 with get, set 
+    member val numJobActionsInProcess = ref 0 with get
+    /// Exception to throw here
+    member x.Exception with get() = if exCollections.IsEmpty then null else AggregateException( exCollections )
```

Comment:
Notice that JobLifeCycle and SingleJobAction are both disposable object. You can't use 
use a = new TypeA()

if TypeA is an option.